ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationIT.scala

```diff
@@ -3,35 +3,997 @@
+import com.daml.ledger.api.testtool.infrastructure.FutureAssertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.ProtobufConverters._
+import com.daml.ledger.api.testtool.infrastructure.participant.{Features, ParticipantTestContext}
+import com.daml.ledger.api.testtool.suites.CommandDeduplicationIT.{
   DelayMechanism,
+  OffsetWithCompletion,
+  StaticTimeDelayMechanism,
   TimeDelayMechanism,
 }
-import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.admin.config_management_service.TimeModel
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.daml.ledger.api.v1.command_submission_service.SubmitRequest
+import com.daml.ledger.api.v1.commands.Commands.DeduplicationPeriod
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.completion.Completion.{
+  DeduplicationPeriod => CompletionDeduplicationPeriod
+}
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.api.v1.version_service.DeduplicationPeriodSupport.OffsetSupport
+import com.daml.ledger.client.binding.Primitive.Party
+import com.daml.ledger.test.model.DA.Types.Tuple2
+import com.daml.ledger.test.model.Test.{Dummy, DummyWithAnnotation, TextKey, TextKeyOperations}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.SubmissionId
+import com.daml.timer.Delayed
+import io.grpc.Status.Code
+import org.slf4j.LoggerFactory
 
-import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
 
-/** Command deduplication tests for participant side deduplication
-  * Should be disabled for ledgers that have committer side deduplication enabled (KV)
-  */
 final class CommandDeduplicationIT(
     timeoutScaleFactor: Double,
-    ledgerTimeInterval: FiniteDuration,
     staticTime: Boolean,
-) extends CommandDeduplicationBase(
-      timeoutScaleFactor,
-      ledgerTimeInterval,
-      staticTime,
-    ) {
-
-  override def runWithDeduplicationDelay(
-      participants: Seq[ParticipantTestContext]
+) extends LedgerTestSuite {
+
+  private[this] val logger = LoggerFactory.getLogger(getClass.getName)
+  val deduplicationDuration: FiniteDuration = scaledDuration(3.seconds)
+
+  test(
+    s"SimpleDeduplicationBasic",
+    "Deduplicate commands within the deduplication duration window",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    val request = ledger
+      .submitRequest(party, DummyWithAnnotation(party, "Duplicate command").create.command)
+      .update(
+        _.commands.deduplicationPeriod :=
+          DeduplicationPeriod.DeduplicationDuration(deduplicationDuration.asProtobuf)
+      )
+    val acceptedSubmissionId1 = newSubmissionId()
```

Comment:
What happened to the second accepted submission?

Reply:
I simplified the test cases as they had a lot of overlap. We now have one more comprehensive test case `DeduplicateMixedClients` which also handles the second submission.